### DISCUSSION TITLE
Code Reviews - Interactive Rebase or Squash Merge

### DISCUSSION THREAD

**Discussion by SudharakaP** [opencollective/opencollective#4025] [Development Q&A] [answered]
@opencollective/core : Hi everyone, this is a question I had for a while. 😄. So when we submit PRs, I notice that some contributors prefer using interactive re-bases with force-pushes; essentially giving the final version a single commit PR. And our document seems to suggest interactive re-bases (https://github.com/opencollective/opencollective-frontend/blob/master/CONTRIBUTING.md). 

But recently it came across me, from my own experience that reviewing a PR is difficult when doing interactive re-bases; after doing the changes suggested and then doing a interactive re-base with force push essentially rewriting the history will make the reviewer go through the changes all over again. Therefore I was naturally opting for making single commits and putting the "Needs Squashing" label on the PR. 

I was wondering what's the best approach. Do you have some preference as a reviewer or any standard we should maintain? 😄

**Comment by Betree** [comment, marked as answer]
Unless specified otherwise in a review both approaches are fine, though it's true that for some kind of iterative work reviewing is much easier when commits are not amended. A frequent strategy is to amend until the PR is ready for review, and push new commits once someone starts reviewing it (and squash at the end). 

I personally have no strong opinion on this topic, but it's open for proposals if you think we should enforce something.

**Reply by SudharakaP** [reply]
Thanks for the comments; much appreciated. So,  keep on doing the squashing approach for myself for larger PRs hereafter unless someone has objections. 😄
